patroni/ha.py

```diff
@@ -242,6 +245,11 @@ def process_sync_replication(self):
                     if not self.dcs.write_sync_state(self.state_handler.name, None, index=self.cluster.sync.index):
                         logger.info('Synchronous replication key updated by someone else.')
                         return
+
+                if self.is_synchronous_mode_strict() and picked is None:
+                    picked = 'patroni_dummy_host'
```

Comment:
I still think it would be better to have a `'*'` there and I will explain why.
First of all let's remember how `synchronous_mode` works:
 1. Patroni picks one of the streaming replicas (closest to the master)
 2. It sets this replica as synchronous in the `postgresql.conf` and runs `pg_ctl reload`
 3. Sleeps for a couple of seconds
 4. Verifies that chosen replica is really became synchronous
 5. Publish this replica into DCS

If the step 4 didn't succeeded it will continue with HA loop and return to the task of choosing and setting synchronous replica only on the next iteration. 

So Patroni doesn't have a goal to block writes on the master until synchronous replica is published in DCS.
During `loop_wait`seconds there is a chance that some other replica will connect, but it will have no chances to become synchronous until the next iteration...

If the `synchronous_standby_names` is set to `'*'`, replica will connect and become synchronous and Patroni will need to go only through steps 1, 2, 5.